Add typescript classes guide. (#26244)
freeCodeCamp/freeCodeCamp: guide/english/typescript/classes/index.md

---
title: Classes
---

# Classes

> `Object Oriented Programming` is worthwhile where tons of code are written, by structuring the code it makes vivid to understand.

Reviewer: Please re-phrase these lines so they are easier to understand.
> `Object Oriented Programming` is worthwhile where tons of code are written, by structuring the code it makes vivid to understand.
Class is one of the paradigm of `OOP`.
TypeScript class reserved `class` keyword to create a class. The syntax is given below - 

Maybe something like this -

```suggestion
Object Oriented Programming is an important programming principle, especially where a lot of code is written. By structuring the code in classes and objects, it makes the code easier to understand.
```

Same with this line -
> TypeScript class reserved `class` keyword to create a class. The syntax is given below -

Class is one of the paradigm of `OOP`.

TypeScript class reserved `class` keyword to create a class. The syntax is given below - 

```typescript
class <class_name>{

>       //Fields

Reviewer: Add a space after the comment and the comment text -
```suggestion
      // Fields
      // Constructors
      // Functions
```

      //Constructors
      //Functions
      }
```
Class can embrace fields, constructors and functions in it.

```typescript
class Pet {  
   name:string; 
 
   constructor(name:string) { 

>       this.name = name 

Reviewer: Add the ending `;` to be consistent. 
```suggestion
      this.name = name; 
```

   }  

   function show():void { 

>       console.log("Pet name is  :   "+this.name) 

Reviewer: Add ending `;`.
```suggestion
      console.log("Pet name is  :   "+this.name); 
```

   } 
}
```
To access the attributes of class `Pet`, It needs to create an instance of the class like under the hood -

```typescript
let pet = new Pet('Bruno');

> pet.show();

Reviewer: Add the expected output in a comment beside this line -
```suggestion
pet.show(); // output goes here
```
